lib/compiler/passes/generate-js.js

```diff
@@ -204,19 +269,24 @@ function generateJS(ast, options) {
 
     return new SourceNode(
       null, null, options.grammarSource, [
-        ast.literals.map(
+        (ast.literals || []).map(
           (c, i) => "  var " + l(i) + " = " + buildLiteral(c) + ";"
-        ).concat("", ast.classes.map(
+        ).concat("", (ast.classes || []).map(
           (c, i) => "  var " + r(i) + " = " + buildRegexp(c) + ";"
-        )).concat("", ast.expectations.map(
+        )).concat("", (ast.expectations || []).map(
           (c, i) => "  var " + e(i) + " = " + buildExpectation(c) + ";"
         )).concat("").join("\n"),
-        ast.functions.map(buildFunc),
+        ...(ast.functions || []).map(buildFunc),
```

Comment:
This was the bug, where SourceNode's type info says the array can contain SourceNodes and strings, but not arrays of SourceNodes.